tools/cli/admin_task_list_commands_test.go

```diff
@@ -553,6 +749,48 @@ func TestAdminUpdateTaskListPartitionConfig(t *testing.T) {
 			numReadPartitions:  2,
 			numWritePartitions: 2,
 		},
+		{
+			name: "Unsafe - one type fails validation",
+			setupMocks: func(client *admin.MockClient, f *frontend.MockClient) {
+				f.EXPECT().DescribeTaskList(gomock.Any(), &types.DescribeTaskListRequest{
+					Domain:       "test-domain",
+					TaskList:     &types.TaskList{Name: "test-tasklist", Kind: types.TaskListKindNormal.Ptr()},
+					TaskListType: types.TaskListTypeDecision.Ptr(),
+				}).Return(&types.DescribeTaskListResponse{
+					Pollers: []*types.PollerInfo{
+						{
+							Identity: "poller",
+						},
+					},
+					PartitionConfig: &types.TaskListPartitionConfig{
+						Version:         1,
+						ReadPartitions:  createPartitions(3),
+						WritePartitions: createPartitions(3),
+					},
+				}, nil).Times(1)
+				f.EXPECT().DescribeTaskList(gomock.Any(), &types.DescribeTaskListRequest{
+					Domain:       "test-domain",
+					TaskList:     &types.TaskList{Name: "test-tasklist", Kind: types.TaskListKindNormal.Ptr()},
+					TaskListType: types.TaskListTypeActivity.Ptr(),
+				}).Return(&types.DescribeTaskListResponse{
+					Pollers: []*types.PollerInfo{
+						{
+							Identity: "poller",
+						},
+					},
+					PartitionConfig: &types.TaskListPartitionConfig{
+						Version:         1,
+						ReadPartitions:  createPartitions(2),
+						WritePartitions: createPartitions(2),
+					},
+				}, nil).Times(1)
+			},
+			expectedError:      "Decision",
```

Comment:
could probably use a more descriptive expected-error, or some kind of comment.  this is rather vague about what it is checking, since "Decision" could be in lots of error messages.